tests/security/389ds/tls_389ds_server.pm

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Implement & Integrate 389ds + sssd test case into openQA,
+#          This test module covers the server setup processes
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#88513
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use mmapi 'wait_for_children';
+
+sub run {
+    select_console("root-console");
+
+    my $local_ip    = '10.0.2.101';
+    my $remote_ip   = '10.0.2.102';
+    my $local_name  = '389ds';
+    my $remote_name = '389dc';
+    my $ca_dir      = '/etc/openldap/ssl';
+    my $inst_ca_dir = '/etc/dirsrv/slapd-localhost';
+
+    # Install 389-ds and create an server instance
+    zypper_call("in 389-ds");
+    assert_script_run("wget --quiet " . data_url("389ds/instance.inf") . " -O /tmp/instance.inf");
+    assert_script_run("dscreate from-file /tmp/instance.inf");
+    validate_script_output("dsctl localhost status", sub { m/Instance.*is running/ });
+
+    # Configure CA Certificates for TLS
+    assert_script_run("wget --quiet " . data_url("389ds/tls_cert.sh") . " -O /tmp/tls_cert.sh");
+    assert_script_run("wget --quiet " . data_url("389ds/.dsrc") . " -O /root/.dsrc");
+    assert_script_run("bash /tmp/tls_cert.sh $local_name $ca_dir");
+    assert_script_run("certutil -D -d $inst_ca_dir -n Server-Cert");
```

Comment:
mozilla-nss-tools is not default installed. I am curious why certutil command can be used directly? Is it a dependency of 389-ds package?

Reply:
@rfan1 can you explain this prior to resolving it?

Reply:
# zypper in 389-ds
Refreshing service 'Basesystem_Module_15_SP3_x86_64'.
Refreshing service 'Desktop_Applications_Module_15_SP3_x86_64'.
Refreshing service 'SUSE_Linux_Enterprise_Server_15_SP3_x86_64'.
Refreshing service 'SUSE_Package_Hub_15_SP3_x86_64'.
Refreshing service 'Server_Applications_Module_15_SP3_x86_64'.
Retrieving repository 'SLE-Module-Basesystem15-SP3-Updates' metadata ............................................[done]
Building repository 'SLE-Module-Basesystem15-SP3-Updates' cache .................................................[done]
Retrieving repository 'SLE-Module-Desktop-Applications15-SP3-Updates' metadata ..................................[done]
Building repository 'SLE-Module-Desktop-Applications15-SP3-Updates' cache .......................................[done]
Retrieving repository 'SLES15-SP3-15.3-0' metadata ..............................................................[done]
Building repository 'SLES15-SP3-15.3-0' cache ...................................................................[done]
Retrieving repository 'SLE-Module-Packagehub-Subpackages15-SP3-Updates' metadata ................................[done]
Building repository 'SLE-Module-Packagehub-Subpackages15-SP3-Updates' cache .....................................[done]
Retrieving repository 'SLE-Module-Server-Applications15-SP3-Updates' metadata ...................................[done]
Building repository 'SLE-Module-Server-Applications15-SP3-Updates' cache ........................................[done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 48 NEW packages are going to be installed:
  389-ds checkpolicy db48-utils krb5-client lib389 libgfortran4 libopenblas_pthreads0 libquadmath0 libsvrcore0
  **mozilla-nss-tools** openldap2-client policycoreutils policycoreutils-lang python3-argcomplete python3-argparse-manpage
  python3-audit python3-Cycler python3-distro python3-fs python3-ipy python3-keyczar python3-keyring
  python3-keyrings.alt python3-kiwisolver python3-ldap python3-matplotlib python3-networkx python3-numpy
  python3-olefile python3-Pillow python3-policycoreutils python3-psutil python3-pyasn1-modules python3-pycryptodome
  python3-pyftpdlib python3-pygraphviz python3-pysendfile python3-python-dateutil python3-pytz python3-pyxdg
  python3-PyYAML python3-scipy python3-SecretStorage python3-selinux python3-semanage python3-setools python3-wheel
  selinux-tools